lib/get/get.js

```diff
@@ -74,7 +74,7 @@ module.exports = function get(walk, isJSONG) {
 
             // If there was a short, then we 'throw an error' to the outside
             // calling function which will onError the observer.
-            if (currentCachePosition && currentCachePosition.$type) {
+            if (!currentCachePosition || currentCachePosition.$type) {
```

Comment:
Can you walk me through the issue that this change resolves?

Reply:
Sure. `currentCachePosition` could become undefined after a bound model's data becomes invalidated. Instead of `model.get` for bound models throwing `TypeError`, this change ensures they throw `InvalidModelError`